include/xtensor/xadapt.hpp

```diff
@@ -100,7 +100,7 @@ namespace xt
      */
     template <class C, std::size_t N>
     xtensor_adaptor<C, N, layout_type::dynamic>
-    xadapt(C& container, const std::array<typename C::size_type, N>& shape, const std::array<typename C::size_type, N>& strides);
+    xadapt(C& container, const stat_shape<typename C::size_type, N>& shape, const stat_shape<typename C::size_type, N>& strides);
```

Comment:
I think in terms of naming i would prefer `xshape` and then decide on it being static vs. dynamic based on the template arguments. Is it necessary to have `dyn_shape` vs `stat_shape`?